pykechain/client.py

```diff
@@ -1341,6 +1343,137 @@ def create_service(self, name, scope, description=None, version=None,
+                     team=None):
+        """
+        Create a Scope.
+
+        :param name: Name of the scope
+        :type name: basestring
+        :param status: choose one of the :class:`enums.ScopeStatus`, defaults to `ScopeStatus.ACTIVE`
+        :type status: basestring or None
+        :param description: (optional) Description of the scope
+        :type description: basestring or None
+        :param tags: (optional) List of tags to be added to the new scope
+        :type tags: list or None
+        :param start_date: (optional) start date of the scope
+        :type start_date: datetime.datetime or None
+        :param due_date: (optional) due date of the scope
+        :type due_date: datetime.datetime or None
+        :param team: Team object
+        :type team :class: `models: Team`
+        :return: the created :class:`models.Scope`
+        :raises APIError: In case of failure of the creation of new Scope
+        """
+        assert isinstance(name, (str, text_type))
+        assert status is None or status in ScopeStatus.values()
+        assert description is None or isinstance(description, (str, text_type))
+        assert tags is None or (isinstance(tags, list) and all([isinstance(t, (str, text_type)) for t in tags]))
+        assert team is None or isinstance(team, Team)
+
+        if not status:
+            status = ScopeStatus.ACTIVE
+        if not start_date:
+            start_date = datetime.datetime.now()
+        if not tags:
+            tags = list()
+
+        data = {
+            'name': name,
+            'status': status,
+            'text': description,
+            'tags': tags,
+            'start_date': start_date,
+            'due_date': due_date,
+            'team_id': team.id if team else None,
+        }
+
+        response = self._request('POST', self._build_url('scopes'),
+                                 data=data)
+
+        if response.status_code != requests.codes.created:  # pragma: no cover
+            raise APIError("Could not create scope, {}:\n\n{}'".format(str(response), response.json()))
+
+        return Scope(response.json()['results'][0], client=self)
+
+    def delete_scope(self, scope):
+        """
+        Delete a scope.
+
+        :param scope: Scope object to be deleted
+        :type scope: :class: `models.Scope`
+
+        :return: None
+        :raises APIError: in case of failure in the deletion of the scope
+        """
+        assert isinstance(scope, Scope), 'Scope "{}" is not a scope!'.format(scope.name)
+
+        response = self._request('DELETE', self._build_url('scope', scope_id=str(scope.id)))
+
+        if response.status_code != requests.codes.no_content:  # pragma: no cover
+            raise APIError("Could not delete scope, {}: {}".format(str(response), response.content))
+
+    def clone_scope(self, scope, name=None, status=None, description=None, tags=None, start_date=None, due_date=None,
```

Comment:
did we check in the PIM2 branch if there is already this code?

incorporate `**kwargs`